web/docs/database/operations/actions.md

```diff
@@ -0,0 +1,889 @@
+import { Task } from "@wasp/entities";
+
+// The 'args' object is the payload sent by the caller (most often from the client)
+export const createTask: CreateTask<Pick<Task, "description">, Task> = async (
+  args,
+  context
+) => {
+  const newTask = await context.entities.Task.create({
+    data: {
+      description: args.description,
+      isDone: false,
+    },
+  });
+  return newTask;
+};
+
+export const markTaskAsDone: MarkTaskAsDone<Pick<Task, "id">, void> = async (
+  args,
+  context
+) => {
+  await context.entities.Task.update({
+    where: { id: args.id },
+    data: { isDone: true },
+  });
+};
+```
+
+Again, annotating the Actions is optional, but greatly improves **full-stack type safety**.
+
+</TabItem>
+</Tabs>
+
+The object `context.entities.Task` exposes `prisma.task` from [Prisma's CRUD API](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/crud).
+
+### Prisma Error Helpers
+
+In your Operations, you may wish to handle general Prisma errors with HTTP-friendly responses.
+
+Wasp exposes two helper functions, `isPrismaError`, and `prismaErrorToHttpError`, for this purpose. As of now, we convert two specific Prisma errors (which we will continue to expand), with the rest being `500`. See the [source here](https://github.com/wasp-lang/wasp/blob/main/waspc/e2e-test/test-outputs/waspMigrate-golden/waspMigrate/.wasp/out/server/src/utils.js).
+
+Here's how you can import and use them:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js
+import { isPrismaError, prismaErrorToHttpError } from "@wasp/utils.js";
+
+// ...
+
+try {
+  await context.entities.Task.create({...})
+} catch (e) {
+  if (isPrismaError(e)) {
+    throw prismaErrorToHttpError(e)
+  } else {
+    throw e
+  }
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```js
+import { isPrismaError, prismaErrorToHttpError } from "@wasp/utils.js";
+
+// ...
+
+try {
+  await context.entities.Task.create({...})
+} catch (e) {
+  if (isPrismaError(e)) {
+    throw prismaErrorToHttpError(e)
+  } else {
+    throw e
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Cache Invalidation
+
+One of the trickiest parts of managing a web app's state is making sure the data returned by the Queries is up to date.
+Since Wasp uses _react-query_ for Query management, we must make sure to invalidate Queries (more specifically, their cached results managed by _react-query_) whenever they become stale.
+
+It's possible to invalidate the caches manually through several mechanisms _react-query_ provides (e.g., refetch, direct invalidation).
+However, since manual cache invalidation quickly becomes complex and error-prone, Wasp offers a quicker and a more effective solution to get you started: **automatic Entity-based Query cache invalidation**.
+Because Actions can (and most often do) modify the state while Queries read it, Wasp invalidates a Query's cache whenever an Action that uses the same Entity is executed.
```

Comment:
```suggestion
To simplify this process, Wasp offers a faster and more effective solution: **automatic Entity-based Query cache invalidation**. Since Actions often modify the state while Queries read it, Wasp automatically invalidates a Query's cache whenever an Action that uses the same Entity is executed.
```